src/ProjectService.Business/Commands/File/FindFilesCommand.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using LT.DigitalOffice.Kernel.BrokerSupport.AccessValidatorEngine.Interfaces;
+using LT.DigitalOffice.Kernel.Constants;
+using LT.DigitalOffice.Kernel.Extensions;
+using LT.DigitalOffice.Kernel.FluentValidationExtensions;
+using LT.DigitalOffice.Kernel.Helpers.Interfaces;
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.Kernel.Validators.Interfaces;
+using LT.DigitalOffice.Models.Broker.Enums;
+using LT.DigitalOffice.Models.Broker.Models.File;
+using LT.DigitalOffice.ProjectService.Broker.Requests.Interfaces;
+using LT.DigitalOffice.ProjectService.Business.Commands.File.Interfaces;
+using LT.DigitalOffice.ProjectService.Data.Interfaces;
+using LT.DigitalOffice.ProjectService.Models.Db;
+using LT.DigitalOffice.ProjectService.Models.Dto.Requests.Filters;
+using Microsoft.AspNetCore.Http;
+
+namespace LT.DigitalOffice.ProjectService.Business.Commands.File
+{
+  public class FindFilesCommand : IFindFilesCommand
+  {
+    private readonly IFileRepository _repository;
+    private readonly IResponseCreator _responseCreator;
+    private readonly IAccessValidator _accessValidator;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IProjectUserRepository _projectUserRepository;
+    private readonly IFileService _fileService;
+    private readonly IBaseFindFilterValidator _findFilterValidator;
+
+    public FindFilesCommand(
+      IFileRepository repository,
+      IResponseCreator responseCreator,
+      IAccessValidator accessValidator,
+      IHttpContextAccessor httpContextAccessor,
+      IProjectUserRepository projectUserRepository,
+      IFileService fileService,
+      IBaseFindFilterValidator findFilterValidator)
+    {
+      _repository = repository;
+      _responseCreator = responseCreator;
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _projectUserRepository = projectUserRepository;
+      _fileService = fileService;
+      _findFilterValidator = findFilterValidator;
+    }
+
+    public async Task<FindResultResponse<FileCharacteristicsData>> ExecuteAsync(FindProjectFilesFilter findFilter)
+    {
+      if (!_findFilterValidator.ValidateCustom(findFilter, out List<string> errors))
+      {
+        return _responseCreator.CreateFailureFindResponse<FileCharacteristicsData>(HttpStatusCode.BadRequest, errors);
+      }
+
+      FileAccessType accessType = FileAccessType.Public;
+
+      DbProjectUser user = (await _projectUserRepository.GetAsync(new List<Guid> { _httpContextAccessor.HttpContext.GetUserId() })).FirstOrDefault();

```

Comment:
do you get all the user projects?
you should get user project by request project id and check the user role

Reply:
done